### ISSUE TITLE
Investigate label-content-name-mismatch when text contains short name forms of days

### ISSUE THREAD

**Issue by straker** [dequelabs/axe-core#4821]
This was reported to us recently. In the following example, the visual text uses a short form for the day name (Wed) while the accessible name of the element uses the full name (Wednesday). This gives a violation for `label-content-name-mismatch`. We'll need to investigate if this is a violation of the WCAG SC, or if not how we would fix it so it doesn't report a problem.

```html
<button aria-label="Wednesday the 14th">Wed the 14th</button>
```

**Comment by chutchins25** [comment]
Investigated this for #5226. Still reports a violation on 4.13.0:

```html
<button aria-label="Wednesday the 14th">Wed the 14th</button>
```

Visible label "Wed the 14th" vs accessible name "Wednesday the 14th" — "Wed" abbreviates "Wednesday", so the visible label isn't contained in the accessible name.

Findings:

1. **Abbreviation cases are an explicit assumption of ACT rule 2ee8b8** — it assumes the visible label doesn't use abbreviations/different spelling from the accessible name. So flagging this is consistent with the rule as written; the abbreviation scenario is a known gap in the underlying SC/ACT rule, not an axe-specific bug.
2. **It's arguably a genuine 2.5.3 failure.** If a speech-input user says the visible "Wed the 14th", it won't match a control whose accessible name is "Wednesday the 14th" — assistive tech keying off the accessible name needs the full form (per @WilcoFiers's Voice Control findings in #5228). Under that lens the violation is correct.
3. **Not reliably fixable in the rule.** Detecting that "Wed" abbreviates "Wednesday" (vs. an unrelated word) needs an abbreviation dictionary and would introduce false negatives — no safe heuristic.

Recommendation: treat as working-as-intended (a true label-in-name concern) rather than suppressing abbreviation mismatches. Open to going more lenient if preferred.

**Comment by chutchins25** [comment]
cc @straker for the call on this one — my recommendation is working-as-intended (details above).

**Comment by WilcoFiers** [comment]
Going to close this issue. Similar problems have been discussed by the W3C and they've concluded that abbreviations like this wouldn't meet the criterion.
